WordPress/Classes/ViewRelated/Blog/BlogListViewController.m

```diff
@@ -136,6 +135,8 @@ - (void)viewWillAppear:(BOOL)animated
     self.resultsController.delegate = self;
     [self.resultsController performFetch:nil];
     [self.tableView reloadData];
+    [self updateEditButton];
+    [self maybeShowNUX];
```

Comment:
Regarding showing NUX, one approach might be to move the logic related to `- (void)removeSelfHostedAccountIfNeeded:(WPAccount*)account context:(NSManagedObjectContext*)context` out of `Blog.m` and into `AccountService.m`, then dispatch a notification when the last account is removed, similar to how `WPAccountDefaultWordPressComAccountChangedNotification` is done, that the AppDelegate could listen to.